homeassistant/components/fan/insteon_local.py

```diff
@@ -31,85 +28,27 @@
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the Insteon local fan platform."""
     insteonhub = hass.data['insteon_local']
+    linked = discovery_info['linked']
```

Comment:
Add a check before this line `if discovery_info is None` and return if that's true. You only want to continue with setup of the platform if it's done via discovery.